azkustoingest/internal/status/status_table_client.go

```diff
@@ -1,67 +1,86 @@
 package status
 
 import (
+	"context"
+	"encoding/json"
 	"github.com/Azure/azure-kusto-go/azkustoingest/internal/resources"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/data/aztables"
 	"github.com/Azure/azure-sdk-for-go/storage"
 	"github.com/google/uuid"
 )
 
 const (
+	// todo: should we set this timeout?
 	defaultTimeoutMsec = 10000
-	fullMetadata       = "application/json;odata=fullmetadata"
+	fullMetadata       = aztables.MetadataFormatFull
 )
 
 // TableClient allows reading and writing to azure tables.
 type TableClient struct {
 	tableURI resources.URI
-	client   storage.Client
-	service  storage.TableServiceClient
-	table    *storage.Table
+	client   *aztables.Client
+
+	service storage.TableServiceClient
+	table   *storage.Table
 }
 
 // NewTableClient Creates an azure table client.
-func NewTableClient(uri resources.URI) (*TableClient, error) {
-	c, err := storage.NewAccountSASClientFromEndpointToken(uri.URL().String(), uri.SAS().Encode())
+func NewTableClient(client policy.Transporter, uri resources.URI) (*TableClient, error) {
+	cred, err := aztables.NewClientWithNoCredential(uri.String(), &aztables.ClientOptions{
+		ClientOptions: azcore.ClientOptions{
+			Transport: client,
+		},
+	})
 	if err != nil {
 		return nil, err
 	}
 
-	ts := c.GetTableService()
-
 	return &TableClient{
 		tableURI: uri,
-		client:   c,
-		service:  ts,
-		table:    ts.GetTableReference(uri.ObjectName()),
+		client:   cred,
 	}, nil
 }
 
-// Read reads a table record cotaining ingestion status.
-func (c *TableClient) Read(ingestionSourceID string) (map[string]interface{}, error) {
+// Read reads a table record containing ingestion status.
+func (c *TableClient) Read(ctx context.Context, ingestionSourceID string) (map[string]interface{}, error) {
 	var emptyID = uuid.Nil.String()
-	entity := c.table.GetEntityReference(ingestionSourceID, emptyID)
+	entity, err := c.client.GetEntity(ctx, ingestionSourceID, emptyID, nil)
+	if err != nil {
+		return nil, err
+	}
 
-	err := entity.Get(defaultTimeoutMsec, fullMetadata, nil)
+	bytes := entity.Value
+	m := make(map[string]interface{})
+	err = json.Unmarshal(bytes, &m)
 	if err != nil {
 		return nil, err
 	}
 
-	return entity.Properties, nil
+	return m, nil
 }
 
-// Write reads a table record cotaining ingestion status.
-func (c *TableClient) Write(ingestionSourceID string, data map[string]interface{}) error {
-	var emptyID = uuid.Nil.String()
-	entity := c.table.GetEntityReference(ingestionSourceID, emptyID)
-	entity.Properties = data
-
-	options := &storage.EntityOptions{}
-	options.Timeout = defaultTimeoutMsec
+// Write reads a table record containing ingestion status.
+func (c *TableClient) Write(ctx context.Context, ingestionSourceID string, data map[string]interface{}) error {
+	dataCopy := make(map[string]interface{})
```

Comment:
Why does this require coping?

Reply:
Technically it's surprising behavior if it modifies the data it received, but since we call it with a map that we throw away anyway there is no point of copying, so I removed the copy.